doc/examples/example5.md

```diff
@@ -0,0 +1,397 @@
+    ...
+    virtual std::vector<uint8_t> signMessage(const std::vector<uint8_t> &message) = 0;
+};
+
+class DigestVerificationCtx {
+    ...
+    virtual void verifyDigest(const std::vector<uint8_t> &signature,
+                              const std::vector<uint8_t> &digest) = 0;
+};
+
+class MessageVerificationCtx {
+    ...
+    virtual void verifyMessage(const std::vector<uint8_t> &signature,
+                               const std::vector<uint8_t> &message) = 0;
+};
+\endcode
+
+#### Example
+If a context supports both, signing messages and message digest, the following code snippets are equal. For simplicity it is assumes that SHA256 is used to create the signed hash.
+\code{.cpp}
+    std::shared_ptr<DigestSignatureCtx> ctx = ...; // Hash Function set to SHA256
+    std::vector<uint8_t> message = {'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd'};
+    std::vector<uint8_t> digest = mococrw::Hash::sha256(message);
+    std::vector<uint8_t> signature = ctx->signDigest(digest);
+\endcode
+\code{.cpp}
+    std::shared_ptr<MessageSignatureCtx> = ...; // Hash Function set to SHA256
+    std::vector<uint8_t> message = {'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd'};
+    std::vector<uint8_t> siganture = ctx->signMessage(message)
+\endcode
+
+The verification contexts behave similarly.
+
+# RSA - Encryption
+
+By default the RSA encryption crypto contexts (for en- and decryption) are using the following options:
+ * OAEP Padding:
+    * Hash Function SHA256
+    * Mask Generation Function: MGF1(SHA256)
+    * Empty label
+
+## Encryption (Default Padding)
+
+\code{.cpp}
+std::string pubKey= R"(-----BEGIN PUBLIC KEY-----
+...
+-----END PUBLIC KEY-----)";
+
+std::vector<uint8_t> encryptedMessage = {...};
+
+mococrw::AsymmetricPublicKey key = mococrw::AsymmetricKeypair::readPublicKeyFromPEM(pubKey);
+
+/*
+ * ...
+ */
+
+mococrw::RSAEncryptionPublicKeyCtx ctx = mococrw::RSAEncryptionPublicKeyCtx(key);
+
+std::vector<uint8_t> encryptedMessage = ctx.encrypt(message)
+\endcode
+
+## Decryption (Default Padding)
+
+\code{.cpp}
+std::string privKey= R"(-----BEGIN PRIVATE KEY-----
+...
+-----END PRIVATE KEY-----)";
+
+std::vector<uint8_t> message = {...};
+
+mococrw::AsymmetricPrivateKey key = mococrw::AsymmetricKeypair::readPrivateKeyFromPEM(privKey, "");
+
+/*
+ * ...
+ */
+
+mococrw::RSAEncryptionPrivateKeyCtx ctx = mococrw::RSAEncryptionPrivateKeyCtx(key);
+
+try {
+    std::vector<uint8_t> decryptedMessage = ctx.decrypt(message);
+}
+catch (const MoCOCrWException &e)  {
+    std::cerr << "Decryption Failed" << std::endl;
+    ...
+}
+\endcode
+
+## Padding Modes
+
+The current RSA encryption interface supports the following types of paddings for encryption:
+* No Padding
+* PKCS v1.5
+* OAEP
+
+### No Padding
+The following code shows how to create that de-/encrypts without padding the content. Please note that this entails the message size equals the size of the RSA key. Thus this mode should only be used with care:
```

Comment:
The comment is not really correct. The message size in the case can be actually smaller than the key size and this is the source of some problems. In contrary, with padding, the message size equals the size of the key.

Even though this is not critical to the code, it might be confusing to the reader and should be corrected.

Reply:
Well, I think it is correct in the context of this library. See comment to NoPadding below...